data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -268,47 +300,22 @@ private Properties getConsumerProperties(final TopicConfig topicConfig) {
             throw new RuntimeException("Bootstrap servers are not specified");
         }
         properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+            if (isKafkaClusterExists(sourceConfig.getBootStrapServers())) {
+                throw new RuntimeException("Can't be able to connect to the given Kafka brokers... ");
```

Comment:
I don't think we should exit if we can't connect. We should retry.

Reply:
Working on this. We create dedicated threads for checking each Kafka Server availability and their connection retry part.